src/main/java/de/tum/in/www1/artemis/service/BuildLogEntryService.java

```diff
@@ -58,4 +59,72 @@ public BuildLogEntryService(BuildLogEntryRepository buildLogEntryRepository, Pro
         }
         return List.of();
     }
+
+    /**
+     * Determines if a given build log string can be categorized as an illegal reflection
+     *
+     * @param logString current build log string
+     * @return boolean indicating an illegal reflection log or not
+     */
+    public boolean isIllegalReflectionLog(String logString) {
+        return logString.startsWith("WARNING") && (logString.contains("An illegal reflective access operation has occurred")
```

Comment:
Maybe replace the lines `logString.startsWith("WARNING")` and do an early exit if  the log isn't a warning:
```
if (!isWarningLog(log))
  return false;

// The rest here
return logString.contains ...
```

Reply:
this is actually a good idea, if all logs needed to start with "[WARNING]" to fall into this category. 
But in this specific case the log either starts with "WARNING..." or it just contains `"to enable warnings of further illegal reflective access operations"` or `"All illegal access operations will be denied in a future release"`

So this change would show the logs.

Reply:
Good point